test/sqlsmith/mzcompose.py

```diff
@@ -0,0 +1,175 @@
+    "must appear in the GROUP BY clause or be used in an aggregate function",
+    "Expected joined table, found",  # Should fix for multi table join
+    "Expected ON, or USING after JOIN, found",  # Should fix for multi table join
+    "but expression is of type",  # Should fix, but only happens rarely
+    "coalesce could not convert type map",  # Should fix, but only happens rarely
+    "operator does not exist: map",  # Should fix, but only happens rarely
+    "result exceeds max size of",  # Seems expected with huge queries
+    "expected expression, but found reserved keyword",  # Should fix, but only happens rarely with subqueries
+    "Expected right parenthesis, found left parenthesis",  # Should fix, but only happens rarely with cast+coalesce
+    "invalid selection: operation may only refer to user-defined tables",  # Seems expected
+]
+
+
+def is_known_error(e: str) -> bool:
+    for known_error in known_errors:
+        if known_error in e:
+            return True
+    return False
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    # parser.add_argument("--max-queries", nargs="?", default=None, type=int)
+    parser.add_argument("--runtime", nargs="?", default=None, type=int)
+    parser.add_argument("--max-joins", nargs="?", default=None, type=int)
+    parser.add_argument("--seed", nargs="?", default=None, type=int)
+    args = parser.parse_args()
+
+    c.up("materialized")
+
+    # Very simple data for our workload
+    c.sql(
+        """
+        CREATE SOURCE tpch
+          FROM LOAD GENERATOR TPCH (SCALE FACTOR 0.00001)
+          FOR ALL TABLES
+          WITH (SIZE = '1');
+
+        CREATE SOURCE auction
+          FROM LOAD GENERATOR AUCTION (SCALE FACTOR 0.01)
+          FOR ALL TABLES
+          WITH (SIZE = '1');
+
+        CREATE SOURCE counter
+          FROM LOAD GENERATOR COUNTER (SCALE FACTOR 0.0001)
+          WITH (SIZE = '1');"""
+    )
+
+    # queries = args.max_queries or 100000
+    runtime = args.runtime or 600
+    max_joins = (
+        args.max_joins or 2
+    )  # https://github.com/MaterializeInc/materialize/issues/2392
+    seed = args.seed or random.randint(0, 2**31 - 1)
+
+    def kill_sqlsmith_with_delay() -> None:
+        time.sleep(runtime)
+        # c.exec("sqlsmith", "bash", "-c", "kill `pidof sqlsmith`")
+        c.kill("sqlsmith", signal="SIGINT")
+
+    killer = Thread(target=kill_sqlsmith_with_delay)
+    killer.start()
+
+    out = c.run(
+        "sqlsmith",
+        # f"--max-queries={queries}",
+        f"--max-joins={max_joins}",
+        f"--seed={seed}",
+        "--log-json",
+        # "--exclude-catalog", # https://github.com/MaterializeInc/materialize/issues/17871
+        f"--target=host=materialized port=6875 dbname=materialize user=materialize",
+        capture=True,
+        check=False,  # We still get back parsable json on failure, so keep going
+    ).stdout
```

Comment:
Since you have `check=False`, consider implementing some rudimentary exit code checking that will fail the entire CI job if sqlsmith fails to run at all , e.g. if it fails to connect to Materialize at all . I hope sqlsmith would use a separate exit code for fatal errors v.s. sql errors.